addons/website_portal/controllers/main.py

```diff
@@ -91,12 +87,32 @@ def details(self, redirect=None, **post):
 
         return request.render("website_portal.details", values)
 
+    def _details_form_fields(self):
+        """Get the expected billing fields in form."""
+        return {
+            "mandatory": {
+                "name",
+                "phone",
+                "email",
+                "street2",
+                "city",
+                "country_id",
+            },
+            "optional": {
+                "zipcode",
+                "state_id",
+                "vat",
+                "street",
```

Comment:
LGFM ! Meanwhile we use a better way to whitelist fields that with a python method.
But it is always better than the original code.